e2e/allocstats/allocstats.go

```diff
@@ -41,10 +40,37 @@ func (tc *BasicAllocStatsTest) BeforeAll(f *framework.F) {
 // TODO(preetha) - add more test cases with more realistic resource utilization
 func (tc *BasicAllocStatsTest) TestResourceStats(f *framework.F) {
 	nomadClient := tc.Nomad()
+	_, err := e2eutil.ListLinuxClientNodes(nomadClient)
```

Comment:
nitpick: reading, I'd worry that this code would be willing to return an empty list of nodes and a nil error, and want to go check the func every time. I'd feel more confident if error was only for failures and empty was just empty, even though that makes the condition longer.

Reply:
I went back and forth on the API a couple of times over that exact thing, because I constantly am digging around in the golang stdlib source to figure that out too. With a vote to change it, I'll do so 😀

Reply:
Done in 5333bf7